src/transform/index.js

```
@@ -108,14 +111,16 @@ export default options => (stream, {validate = true, fix = true} = {}) => {
 
             /* istanbul ignore next: No tests without validators */ emitter.emit('record', record);
           } catch (err) {
-            if (err.message === 'Unsupported product identifier type & value') {
+
+            if (err instanceof NotSupportedError) {
```

You need no rethrow the error after this block. Otherwise the error is ignored.